File: app/src/main/java/com/keelim/nandadiagnosis/mainfragment/search/db/DatabaseAdapter.java
package com.keelim.nandadiagnosis.mainfragment.search.db;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.keelim.nandadiagnosis.R;

import java.util.List;

public class DatabaseAdapter extends BaseAdapter {
    private Context context;
    private List<DatabaseItem> arrayList;

    public DatabaseAdapter(Context context, List<DatabaseItem> arrayList) {
        this.context = context;
        this.arrayList = arrayList;
    }

    @Override
    public int getCount() {
        return arrayList.size();
    }

    @Override
    public Object getItem(int i) {
        return arrayList.get(i);
    }

    @Override
    public long getItemId(int i) {
        return i;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        if (view == null)
            view = LayoutInflater.from(context).inflate(R.layout.item_db, null); //todo?
        TextView diagnosis_textView = view.findViewById(R.id.db_diagnosis);
        diagnosis_textView.setText(arrayList.get(i).getDiagnosis());
        return view;
    }
}

File: app/src/main/java/com/keelim/nandadiagnosis/model/DatabaseAdapter.kt
package com.keelim.nandadiagnosis.model

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import com.keelim.nandadiagnosis.R
import com.keelim.nandadiagnosis.model.roomdb.NandaEntity

class DatabaseAdapter(private val context: Context, private val arrayList: List<NandaEntity>) : BaseAdapter() {
    override fun getCount(): Int {
        return arrayList.size
    }

    override fun getItem(i: Int): Any {
        return arrayList[i]
    }

    override fun getItemId(i: Int): Long {
        return i.toLong()
    }


    override fun getView(i: Int, view: View?, viewGroup: ViewGroup): View {
        val view = LayoutInflater.from(context).inflate(R.layout.item_db, null)
        view.findViewById<TextView>(R.id.db_diagnosis).text = arrayList[i].diagnosis
        return view
    }


}